### DISCUSSION TITLE
Is there a way to set 'dedupe unit size' other than blocksize?

### DISCUSSION THREAD

**Discussion by cabezadevaca** [axboe/fio#1800] [Q&A]
Does FIO use the blocksize for the dedupe? Can it be set to block size other than 'blocksize'?

**Comment by sitsofe** [comment]
Hello @cabezadevaca:

> Does FIO use the blocksize for the dedupe?

Yes. See:
- The [`io_u_fill_buffer()` call in `get_io_u()`](https://github.com/axboe/fio/blob/fio-3.40/io_u.c#L1910-L1912)
- [`io_u_fill_buffer()` takes `min_write` and `max_bs` which it passes to `fill_io_buffer()`](https://github.com/axboe/fio/blob/fio-3.40/io_u.c#L2420-L2425) along with `io_u->buf`
- In [`fill_io_buffer()`](https://github.com/axboe/fio/blob/fio-3.40/io_u.c#L2378-L2408) `buf` is filled up with dedupable data and random data up to `max_bs`

Even [`init_dedupe_working_set_seeds()` works in sizes of `min_bs`](https://github.com/axboe/fio/blob/master/dedupe.c#L38).

> Can it be set to block size other than 'blocksize'?

No.
